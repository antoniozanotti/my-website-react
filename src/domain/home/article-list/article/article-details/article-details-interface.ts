import { Link } from "@/types/link";

export interface ArticleDetailsInterface extends React.ComponentProps<"div"> {
  company: string;
  location?: string;
  period: string;
  skills?: string[];
}
