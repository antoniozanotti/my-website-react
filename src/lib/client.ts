import { ApolloClient, HttpLink, InMemoryCache } from "@apollo/client";
import { registerApolloClient } from "@apollo/experimental-nextjs-app-support/rsc";

const TOKEN = process.env.CONTENTFUL_ACCESS_TOKEN;
const SPACE = process.env.CONTENTFUL_SPACE_ID;
const ENVIRONMENT = process.env.CONTENTFUL_ENVIRONMENT_ID;
const URL = `https://graphql.contentful.com/content/v1/spaces/${SPACE}/environments/${ENVIRONMENT}`;

export const { getClient } = registerApolloClient(() => {
  return new ApolloClient({
    cache: new InMemoryCache(),
    link: new HttpLink({
      uri: URL,
      headers: {
        Authorization: `Bearer ${TOKEN}`,
      },
    }),
  });
});
