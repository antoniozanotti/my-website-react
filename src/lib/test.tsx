import React, { ReactElement } from "react";
import {
  render,
  RenderOptions,
  RenderResult,
  Screen,
} from "@testing-library/react";
import { Theme } from "@/domain/theme/theme/theme";
import { Providers } from "../app/providers";
import messages from "@/messages/en.json";

function AllTheProviders({ children }: { children: React.ReactNode }) {
  return (
    <Providers locale="en" messages={messages}>
      <Theme>
        <div data-testid="container">{children}</div>
      </Theme>
    </Providers>
  );
}

function customRender(
  ui: ReactElement,
  options?: Omit<RenderOptions, "wrapper">
): RenderResult {
  return render(ui, { wrapper: AllTheProviders, ...options });
}

function firstComponent(screen: Screen) {
  return screen.getByTestId("container").firstChild;
}

export * from "@testing-library/react";
export * from "@testing-library/user-event";
export { customRender as render, firstComponent };
import "@testing-library/jest-dom";

jest.mock("next/navigation", () => {
  return {
    useRouter() {
      return { push: () => {} };
    },
    useParams: () => ({}),
  };
});
