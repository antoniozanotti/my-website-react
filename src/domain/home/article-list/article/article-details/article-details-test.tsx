import { test, expect } from '@playwright/experimental-ct-react';
import { ArticleDetails } from "./article-details";

test.describe("ArticleDetails", () => {
  test("should render default props", async ({ mount }) => {
    const component = await mount(<ArticleDetails company='Company' period='Period' />);
    await expect(component.locator(":scope:is(div)")).toHaveCount(1);
    await expect(component).toContainText("Company");
    await expect(component).toContainText("Period");
  });

  test("should accept location", async ({ mount }) => {
    const component = await mount(<ArticleDetails company='Company' period='Period' className='test' location='Location' />);
    await expect(component).toContainText("Location");
  });

  test("should accept skills", async ({ mount }) => {
    const component = await mount(<ArticleDetails company='Company' period='Period' className='test' skills={["Skill1", "Skill2"]} />);
    await expect(component).toContainText("Skill1");
    await expect(component).toContainText("Skill2");
  });

  test("should accept className", async ({ mount }) => {
    const component = await mount(<ArticleDetails company='Company' period='Period' className='test' />);
    await expect(component).toHaveClass(/test$/);
  });
});
