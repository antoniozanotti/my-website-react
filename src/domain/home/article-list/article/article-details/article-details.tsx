import { ArticleDetailsInterface } from "./article-details-interface";
import { TzTag } from "topaz-react";

export function ArticleDetails({
  company,
  location,
  period,
  skills,
  ...props
}: ArticleDetailsInterface) {
  const baseStyles = "grid grid-flow-row auto-rows-max";
  props.className = props.className
    ? `${baseStyles} ${props.className}`
    : baseStyles;

  return (
    <div {...props}>
      <span>{company}</span>
      {location && (
        <span className="italic text-c2 lg:text-c3">{location}</span>
      )}
      <span className="font-bold py-4">{period}</span>
      {skills && (
        <div className="flex flex-wrap gap-2">
          {skills.map((skill, index) => (
            <TzTag key={index} label={skill} variant="secondary" />
          ))}
        </div>
      )}
    </div>
  );
}
