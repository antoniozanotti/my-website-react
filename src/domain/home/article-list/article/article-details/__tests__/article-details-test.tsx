import { screen, render, firstComponent } from "@/lib/test";
import { ArticleDetails } from "../article-details";

describe("ArticleDetails", () => {
  it("should render default props", async () => {
    const { container } = render(
      <ArticleDetails company="Company" period="Period" />
    );
    const component = firstComponent(screen);
    expect(component).toBeVisible();
    expect(component).toHaveTextContent("Company");
    expect(component).toHaveTextContent("Period");
    expect(container).toMatchSnapshot();
  });

  it("should accept location", async () => {
    const { container } = render(
      <ArticleDetails
        company="Company"
        period="Period"
        className="test"
        location="Location"
      />
    );
    const component = firstComponent(screen);
    expect(component).toHaveTextContent("Location");
    expect(container).toMatchSnapshot();
  });

  it("should accept skills", async () => {
    const { container } = render(
      <ArticleDetails
        company="Company"
        period="Period"
        className="test"
        skills={["Skill1", "Skill2"]}
      />
    );
    const component = firstComponent(screen);
    expect(component).toHaveTextContent("Skill1");
    expect(component).toHaveTextContent("Skill2");
    expect(container).toMatchSnapshot();
  });

  it("should accept className", async () => {
    const { container } = render(<ArticleDetails company="Company" period="Period" className="test" />);
    const component = firstComponent(screen);
    expect(component).toHaveClass(/test$/);
    expect(container).toMatchSnapshot();
  });
});
