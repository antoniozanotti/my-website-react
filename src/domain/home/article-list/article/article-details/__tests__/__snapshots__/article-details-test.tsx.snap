// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ArticleDetails should accept className 1`] = `
<div>
  <div
    class="dark"
  >
    <div
      class="className bg-dark-2 dark:bg-dark-dark-2 text-primary dark:text-dark-primary text-c3 lg:text-c4"
    >
      <div
        data-testid="container"
      >
        <div
          class="grid grid-flow-row auto-rows-max test"
        >
          <span>
            Company
          </span>
          <span
            class="font-bold py-4"
          >
            Period
          </span>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`ArticleDetails should accept location 1`] = `
<div>
  <div
    class="dark"
  >
    <div
      class="className bg-dark-2 dark:bg-dark-dark-2 text-primary dark:text-dark-primary text-c3 lg:text-c4"
    >
      <div
        data-testid="container"
      >
        <div
          class="grid grid-flow-row auto-rows-max test"
        >
          <span>
            Company
          </span>
          <span
            class="italic text-c2 lg:text-c3"
          >
            Location
          </span>
          <span
            class="font-bold py-4"
          >
            Period
          </span>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`ArticleDetails should accept skills 1`] = `
<div>
  <div
    class="dark"
  >
    <div
      class="className bg-dark-2 dark:bg-dark-dark-2 text-primary dark:text-dark-primary text-c3 lg:text-c4"
    >
      <div
        data-testid="container"
      >
        <div
          class="grid grid-flow-row auto-rows-max test"
        >
          <span>
            Company
          </span>
          <span
            class="font-bold py-4"
          >
            Period
          </span>
          <div
            class="flex flex-wrap gap-2"
          >
            <span
              class="rounded font-medium px-2 py-1 text-c2  bg-secondary dark:bg-dark-secondary text-inverse-secondary dark:text-inverse-dark-secondary "
            >
              Skill1
            </span>
            <span
              class="rounded font-medium px-2 py-1 text-c2  bg-secondary dark:bg-dark-secondary text-inverse-secondary dark:text-inverse-dark-secondary "
            >
              Skill2
            </span>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`ArticleDetails should render default props 1`] = `
<div>
  <div
    class="dark"
  >
    <div
      class="className bg-dark-2 dark:bg-dark-dark-2 text-primary dark:text-dark-primary text-c3 lg:text-c4"
    >
      <div
        data-testid="container"
      >
        <div
          class="grid grid-flow-row auto-rows-max"
        >
          <span>
            Company
          </span>
          <span
            class="font-bold py-4"
          >
            Period
          </span>
        </div>
      </div>
    </div>
  </div>
</div>
`;
